WIP Task 4

AlfiyaRF opened the pull request:
Hi guys,

Please give me a few days to complete the task. This week I was on vacation and had only half a day for homework.
If it's convenient for you, you can message me on Teams just to let me know who you are and I'll ping you when I'm done.

Thank you)

aamialkovich commented on product-service/db/insertProduct.js:
```diff
@@ -0,0 +1,43 @@
+import {Client} from 'pg';
+
+const {
```
Looks like we can move it into sepate file and import in all db functions.

aamialkovich commented on product-service/src/getProductsList.js:
```diff
@@ -0,0 +1,39 @@
+        const data = await selectAllProducts();
+        if (data.status === 200) {
+            return {
+                statusCode: data.status,
```
Response can be moved to particular function to reduce duplication.

nullpr commented:
1. please provide a link to the CloudFront,
2. https://rx3w8zwdt0.execute-api.eu-west-1.amazonaws.com/dev/products/f410f55a-7af9-4e61-812d-9e426e1cd612
returns an empty array, but should return 404 (product not found)

3. post request failed with 502
<img width="1002" alt="image" src="https://user-images.githubusercontent.com/3364804/181710572-6d725970-0a1e-4861-880e-47379b20a6b7.png">

AlfiyaRF commented:
> 1. please provide a link to the CloudFront,
> 2. https://rx3w8zwdt0.execute-api.eu-west-1.amazonaws.com/dev/products/f410f55a-7af9-4e61-812d-9e426e1cd612
>    returns an empty array, but should return 404 (product not found)
> 3. post request failed with 502
> 
> <img alt="image" width="1002" src="https://user-images.githubusercontent.com/3364804/181710572-6d725970-0a1e-4861-880e-47379b20a6b7.png">

Thank you for your comments
1. here it is https://d2ocjv21yccmuj.cloudfront.net/
2. fixed it
3. I got the 400 error with the same request, because the price was missed. I added the correct message

viktordavidovich commented:
Well done!
I found few minor issues with _createProduct_ API 
- it returns nothing, technically speaking I'm expecting to get back id of the created product and HTTP status code 204 Created, not 200 OK.
- if I send _price_ or _count_ values as decimals, it still creating a product while doing digital round, which is not really best way to handle it. I'd expect getting back an error.

The max score for this task is 9, but I'd give you only 8.
